ddtrace/opentracer/tracer_test.go

```diff
@@ -37,3 +38,25 @@ func TestSpanWithContext(t *testing.T) {
 	assert.True(ok)
 	assert.Equal(got, want.(*span).Span)
 }
+
+func TestTranslateError(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		assert.Nil(t, translateError(nil))
+	})
+	t.Run("unrecognized", func(t *testing.T) {
+		err := errors.New("unrecognized")
+		assert.Equal(t, err, translateError(err))
+	})
+	t.Run("ErrSpanContextNotFound", func(t *testing.T) {
+		assert.Equal(t, opentracing.ErrSpanContextNotFound, translateError(tracer.ErrSpanContextNotFound))
+	})
+	t.Run("ErrInvalidCarrier", func(t *testing.T) {
+		assert.Equal(t, opentracing.ErrInvalidCarrier, translateError(tracer.ErrInvalidCarrier))
+	})
+	t.Run("ErrInvalidSpanContext", func(t *testing.T) {
+		assert.Equal(t, opentracing.ErrInvalidSpanContext, translateError(tracer.ErrInvalidSpanContext))
+	})
+	t.Run("ErrSpanContextCorrupted", func(t *testing.T) {
+		assert.Equal(t, opentracing.ErrSpanContextCorrupted, translateError(tracer.ErrSpanContextCorrupted))
+	})
+}
```

Comment:
I think this would be better as a table test ([example](https://github.com/DataDog/dd-trace-go/blob/9a5c7ca33b0d8f1b2fd8dacb61fad05875c478cd/ddtrace/tracer/transport_test.go#L109)).

That will reduce the boilerplate and make it simpler to update if we ever add more errors.

Reply:
Refactored tests to be table tests and test that `Inject` and `Extract` actually return the correct error.